src/repl/scala/tools/nsc/interpreter/InteractiveReader.scala

```diff
@@ -23,6 +23,7 @@ trait InteractiveReader {
   def readYesOrNo(prompt: String, alt: => Boolean): Boolean = readOneKey(prompt) match {
     case 'y'  => true
     case 'n'  => false
+    case -1   => false // EOF
```

Comment:
Woo-hoo! Just yesterday this was annoying me again, though with a different crash, and I resolved to fix it. Once upon a time, I also had it take 'q' to quit. I've been rockin' the ctl-D since it started echoing quit, which is the signal that your desire to return to the shell prompt has been acknowledged.